src/lib.rs

```diff
@@ -1877,10 +1873,7 @@ impl Build {
             for flag in self.ar_flags.iter() {
                 ar.arg(flag);
             }
-            run(
-                ar.arg("crs").arg(dst).args(&objects).args(&self.objects),
```

Comment:
I think this may be accidentally dropping `self.object`? That may no longer get linked in anywhere?

Reply:
`self.objects` is now cloned and chained together with the objects provided to the assembler in the new `assemble` method so should not be dropped.

Reply:
Ah oops I looked for it and failed to see that it was in front of my face, thanks though!